### PR TITLE
feat: deprecate Icon Trimmed Variants

### PR DESCRIPTION
**Pull request by Lelith**
# Purpose of PR

Deprecate the trimmed Icon variant, they will be removed with the next Iteration over Icons.

image

### CODE REVIEW

File: packages/components/icon/examples/AllIconsExample.tsx [hunk 1]
```diff
@@ -1,29 +1,72 @@
...
+  Box,
+} from '@contentful/f36-components';
 
 export default function AllIconsExample() {
-  return (
-    <Grid columns="repeat(2, 1fr)">
-      {Object.entries(f36icons).map(([name, icon]) => {
-        const Component = icon;
+  console.log(f36icons);
+
```
Comment (denkristoffer) [inline packages/components/icon/examples/AllIconsExample.tsx:15]:
```suggestion
```

File: packages/components/icon/src/Icon.tsx [hunk 2]
```diff
@@ -74,6 +74,7 @@ export type IconInternalProps = CommonProps & {
   size?: IconSize;
   /**
    * Whether or not to trim the icon width, i.e. set `width` to `auto`
+   * @deprecated IconsV5 will no longer support the trimmed option.
```
Comment (denkristoffer) [inline packages/components/icon/src/Icon.tsx:77]:
```suggestion
   * @deprecated Trimmed icons will be removed in a future major release
```

I think we should be less specific here, haha 😅

**Review by denkristoffer — COMMENTED** [review]
Looks good to me, I wonder if we should discuss this console warning a bit more before merging it. It would bring a lot of noise to our own products, but I'm not against having it per se. It might be too early to bring in such a warning though?

**Comment by Lelith** [comment]
@denkristoffer yeah, i think the warning could become something like a "phase 2" and probaply only a short time period after we did our first migration shot but when other projects might still be using the old ones.

**Review by damann — APPROVED** [review]

**Review by denkristoffer — APPROVED** [review]

**Pull request merged**